credentialAtomicQueryMTP.go

```diff
@@ -211,3 +211,27 @@ func (ao *AtomicQueryMTPPubSignals) PubSignalsUnmarshal(data []byte) error {
 func (ao AtomicQueryMTPPubSignals) GetObjMap() map[string]interface{} {
 	return toMap(ao)
 }
+
+func (ao *AtomicQueryMTPPubSignals) VerifyStates(ctx context.Context, stateVerFunc StateResolverFunc) error {
+
+	userStateVerificationRes, err := stateVerFunc(ctx, ao.UserID.BigInt(), ao.UserState.BigInt())
+	if err != nil {
+		return err
+	}
+
+	if !userStateVerificationRes.Latest {
+		return errUserStateIsNotValid
+	}
+
+	issuerClaimState, err := stateVerFunc(ctx, ao.IssuerID.BigInt(), ao.IssuerClaimIdenState.BigInt())
+	if err != nil {
+		return err
+	}
+	if issuerClaimState == nil {
+		return errIssuerClaimStateIsNotValid
+	}
+
+	//TODO: add revocation state check
```

Comment:
Not clear what is the "revocation state check" for the state... Was this comment just copy-pasted from somewhere by mistake?

Reply:
Because it's logical to check the state of the issuer when revocation mtp was created. Because if it's ok to have the issuer state for claim issuance that is 1 year old.  And for revocation it'ok to use 1 week state old.